examples/llm/src/models/mosaic_gpt.py

```diff
@@ -220,20 +257,51 @@ def fsdp_wrap_fn(self, module):
     def activation_checkpointing_fn(self, module):
         return isinstance(module, gpt_blocks.GPTBlock)
 
+    def prepare_inputs_for_generation(self,
+                                      input_ids,
+                                      past_key_values=None,
+                                      inputs_embeds=None,
+                                      **kwargs):
+        if inputs_embeds is not None:
+            raise NotImplementedError(
+                'inputs_embeds is not implemented for MosaicGPT yet')
+
+        attention_mask = kwargs['attention_mask'].bool()
+        if attention_mask[:, -1].sum() != attention_mask.shape[0]:
+            raise NotImplementedError(
+                'MosaicGPT does not support generation with right padding.')
+
+        if past_key_values is not None:
+            input_ids = input_ids[:, -1].unsqueeze(-1)
```

Comment:
`if past_key_values is not None:` is it safe to assume `input_ids` is of shape B instead of Bx1? or BxSq?

Reply:
Do you mean the input ids that are passed in here? I don't believe it is safe to assume that. Afaik this prepare function is the huggingface way to trim the inputs to a single token.